include ollama open web uimanual (#117)
UtrechtUniversity/vre-docs: docs/workspaces/utility/ollama-open-webui.qmd

gpu: true
admin: false
support: UU
---

## Description

[Open WebUI](https://docs.openwebui.com/) is a user-friendly and privacy-friendly AI platform that allows you to interact with AI models locally on your workspace. This workspace is configured to use [Ollama](https://ollama.com/) models that are downloaded directly to the workspace. You can also use models from [HuggingFace](https://huggingface.co/models). 

Ollama Open WebUI gives you **complete privacy** as all processing happens on your workspace, no information is sent to external AI services.

For security reasons, the Open WebUI API is disabled by default.

## Creation

::: {.callout-important}
## Planning Your Workspace

Before creating your workspace, consider:

1. **Which model do you want to use?** (See recommendations below)
2. **How much storage do you need?** Models can be very large (1GB - 40GB+)
3. **Do you need GPU acceleration?** Required for large models (>8B parameters)

**If you plan to download large models, create and attach a storage volume first.** 

See the [Getting started](../../first-steps.qmd#create-storage-volume) page for more info about how and why to create a storage volume.
:::

> Remember that you are responsible for making backups of any data yourself! Although data on a storage unit will not be deleted when you delete the workspace, it can still become corrupt. So if it is important to your work, make sure to make periodical (e.g. daily or weekly) backups of your projects.

Reviewer: ```suggestion
Remember that you are responsible for making backups of any data yourself! Although data on a storage unit will not be deleted when you delete the workspace, storage volumes are not backed up. Make sure to regularly backup your files, especially files or edits to code that are not easy to recreate when lost.
```

### Recommended Models and Resources

> Here are some popular Ollama models with their resource requirements:

Reviewer: I think this table needs some revision. From what I know: for fast performance users can choose models up to 20b parameters if they use an A10 GPU. gpt-oss 20b is maybe the best option to add here as maximum. 
I think it would be good to have a bit more variation (include e.g. qwen, mistral, gemini options). 
The recommendations for CPU/GPU are mostly fine to my feeling, did you run some tests? Would it be possible to run some tests (chat, text extractions?) We can help define tests if needed. Or this one is maybe nice: https://bit.ly/extractive-ai

Reviewer 2: @ShaivanBhagat let me know how you'd like to continue with this: we can merge this PR as-is, and create an issue to update the table. Or if you're currently working on Jelle's suggestion, we can also just leave this PR open a bit longer. Whatever you prefer!

Author: @dometto I have updated the table. However, Jelle will still be reviewing some additional parts of the manual once he is back. So we can leave it open until then. I see that API usage can also be included in this manual, so I will try to document that part here as well.

| Model | Size | CPU/GPU | Best For |
|-------|------|---------|----------|
| `llama3.2:1b` | ~1 GB |  CPU | Testing, simple tasks, very fast responses |
| `llama3.2` | ~2 GB |  CPU | General chat, quick responses |
| `mistral` | ~4 GB | CPU/ GPU | Balanced speed and quality |
| `llama3.1:8b` | ~5 GB | GPU recommended | High-quality responses, coding |
| `llama3.1:70b` | ~40 GB |  GPU required | Best quality, complex reasoning |

Find all available Ollama models here: [List of Ollama models](https://ollama.com/library)

### Workspace Size Selection

When creating your workspace, you'll need to select a processor configuration:

**Since Ollama is resource-intensive, we recommend:**

> - **For small models (1B-8B):** 2-8 CPU cores or higher

Reviewer: ```suggestion
- **For small models (1B-8B):** 2-16 CPU cores, depending on size
```

> - **For large models (13B+):** GPU workspaces 

Reviewer: ```suggestion
- **For large models (10B+):** GPU workspaces 
```

> Although larger models and GPUs are avaiable, you should not simply select the most hardware available (see [responsible use](../../responsible-use.qmd)).

Reviewer: ```suggestion
These are rough estimations, it may help to check the requirements of the model you would like to use online.
Additionally, to spend energy and credits wisely, it is useful to experiment a bit with different (small to large) workspace configurations to find the appropriate configuration where your pipeline runs satisfactory (see [responsible use](../../responsible-use.qmd)).
```

:::

{{< include ../partials/_creation.qmd >}}

## Access

This workspace can be accessed via the yellow 'Access' button, or by opening the URL listed in the dashboard in your browser. Any member of the collaboration can login to the workspace [using Single-Sign On](../../first-steps.qmd#webapplications-that-support-single-sign-on.).

## Usage

### Pulling Models from Ollama

When you access the workspace for the first time, a default model `smollm:135m` is already loaded. However, you may want to download and use other models. 
See [Ollama models](https://ollama.com/library) for a complete list of models. 

To use a different model, simply copy the name of the desired model and select the model dropdown in the top left. From there, you can search for the required model and pull it. 
Pulling a model may take some time depending on the size of the model. Once finished, you can use the pulled model on your workspace. 

![](../imgs/ollama_pull-model.png)

::: {.callout-note}
Model downloads are stored on your workspace. If you attached a storage volume, they'll be preserved even if you delete and recreate the workspace.

> **NOTE: Cloud-only models CANNOT be pulled**.

Reviewer: ```suggestion
**NOTE: Cloud-only models CANNOT be pulled, and are not recommended when data cannot be send to online services**.
```
